packages/stryker/src/initializer/StrykerInitializer.ts

```diff
@@ -0,0 +1,132 @@
+import { StrykerInquirer } from './StrykerInquirer';
+import NpmClient from './NpmClient';
+import PromptOption from './PromptOption';
+import * as log4js from 'log4js';
+import { filterEmpty } from '../utils/objectUtils';
+import StrykerConfigWriter from './StrykerConfigWriter';
+
+const log = log4js.getLogger('StrykerInitializer');
+
+export default class StrykerInitializer {
+
+  private inquirer = new StrykerInquirer();
+
+  constructor(private out = console.log, private client: NpmClient = new NpmClient()) { }
+
+  /**
+   * Runs the initializer ask used framework and test runner en setup environment
+   * @function
+   */
+  async initialize(): Promise<void> {
+    StrykerConfigWriter.guardForExistingConfig();
+    this.patchProxies();
+    const selectedTestRunner = await this.selectTestRunner();
+    const selectedTestFramework = selectedTestRunner ? await this.selectTestFramework(selectedTestRunner) : null;
+    const selectedReporters = await this.selectReporters();
+    const npmDependencies = this.getSelectedNpmDependencies(selectedTestRunner, selectedTestFramework, selectedReporters);
+    this.installNpmDependencies(npmDependencies);
+    await new StrykerConfigWriter(this.out,
+      selectedTestRunner,
+      selectedTestFramework,
+      selectedReporters,
+      await this.fetchAdditionalConfig(npmDependencies)).write();
+    this.out('Done configuring stryker. Please review `stryker.conf.js`, you might need to configure your files and test runner correctly.');
+    this.out('Let\'s kill some mutants with this command: `stryker run`');
+  }
+
+  /**
+  * The typed rest client works only with the specific HTTP_PROXY and HTTPS_PROXY env settings.
+  * Let's make sure they are available.
+  */
+  private patchProxies() {
+    const copy = (from: string, to: string) => {
+      if (process.env[from] && !process.env[to]) {
+        process.env[to] = process.env[from];
+      }
+    };
+    copy('http_proxy', 'HTTP_PROXY');
+    copy('https_proxy', 'HTTPS_PROXY');
+  }
+
+  private async selectTestRunner(): Promise<PromptOption | null> {
+    const testRunnerOptions = await this.client.getTestRunnerOptions();
+    if (testRunnerOptions.length) {
+      log.debug(`Found test runners: ${JSON.stringify(testRunnerOptions)}`);
+      return await this.inquirer.promptTestRunners(testRunnerOptions);
+    } else {
+      this.out('Unable to select a test runner. You will need to configure it manually.');
+      return null;
+    }
+  }
+
+  private async selectReporters(): Promise<PromptOption[]> {
+    let reporterOptions: PromptOption[];
+    try {
+      reporterOptions = await this.client.getTestReporterOptions();
+    } catch (err) {
+      this.out('Unable to fetch additional reporters.');
+      reporterOptions = [];
+    }
+    reporterOptions.push({
+      name: 'clear-text',
+      npm: null
+    }, {
+        name: 'progress',
+        npm: null
+      });
+    return this.inquirer.promptReporters(reporterOptions);
```

Comment:
It's not needed, that would unwrap the result and wrap it in a new promise. Much easier to just return the promise.